src/replicate_stage.rs

```diff
@@ -48,13 +47,19 @@ impl ReplicateStage {
         }
         let blobs_len = blobs.len();
         let entries = ledger::reconstruct_entries_from_blobs(blobs.clone())?;
-        let votes = entries_to_votes(&entries);
-
-        static mut COUNTER_REPLICATE: Counter = create_counter!("replicate-transactions", LOG_RATE);
-        inc_counter!(
-            COUNTER_REPLICATE,
-            entries.iter().map(|x| x.transactions.len()).sum()
-        );
+        {
+            let votes = entries_to_votes(&entries);
+            let mut wcrdt = crdt.write().unwrap();
+            wcrdt.insert_votes(&votes);
```

Comment:
I don't grok why `insert_votes()` was moved in this patch.

Reply:
Good point. I don't either. @aeyakovenko, there should be a unit test that demonstrates why this side-effect needs to take place outside the condition.